src/Sylius/Behat/Context/Api/Admin/ManagingShippingMethodsContext.php

```diff
@@ -63,6 +64,94 @@ public function iSortShippingMethodsByName(string $sortType = 'ascending'): void
         ]);
     }
 
+    /**
+     * @When I add the :rule rule configured with :weight
+     */
+    public function iAddTheRuleConfiguredWithWeight(string $rule, int $weight): void
+    {
+        match ($rule) {
+            'Total weight less than or equal' => $this->client->addRequestData('rules', [
+                [
+                    'type' => 'total_weight_less_than_or_equal',
+                    'configuration' => [
+                        'weight' => $weight,
+                    ],
+                ],
+            ]),
+            'Total weight greater than or equal' => $this->client->addRequestData('rules', [
+                [
+                    'type' => 'total_weight_greater_than_or_equal',
+                    'configuration' => [
+                        'weight' => $weight,
+                    ],
+                ],
+            ]),
+            default => throw new \InvalidArgumentException('Unsupported shipping method rule'),
+        };
+    }
+
+    /**
+     * @When I add the "Total weight greater than or equal" rule configured with invalid data
+     */
+    public function iAddTheTotalWeightGreaterThanOrEqualRuleConfiguredWithInvalidData(): void
+    {
+        $this->client->addRequestData('rules', [
+            [
+                'type' => 'total_weight_greater_than_or_equal',
+                'configuration' => [
+                    'weight' => true,
+                ],
+            ],
+        ]);
+    }
+
+    /**
+     * @When /^I add the ([^"]+) rule configured with (?:€|£|\$)([^"]+) for ("[^"]+" channel)$/
+     */
+    public function iAddTheRuleConfiguredWithForChannel(string $rule, int $value, ChannelInterface $channel): void
+    {
+        match ($rule) {
+            'Items total less than or equal' => $this->client->addRequestData('rules', [
+                [
+                    'type' => 'order_total_less_than_or_equal',
```

Comment:
Sidenote;
We really need to refactor this, the label has the context of what is actually taken into account, while the type is just wrong. The checker compares the items' total instead of the order's